### PR TITLE
fix(grainfmt): Remove parens around annotated types

### LINKED ISSUE #1107: Update formatter to not put parens around annotated types and reformat stdlib
After #1033 lands, #783 is fixed and we can resolve this.

### PR DESCRIPTION
**Pull request by ospencer**
Fixes #1107

### CODE REVIEW

File: compiler/test/formatter_outputs/application.gr [hunk 1]
```diff
@@ -54,26 +54,26 @@ let ret = filteri(
...
 
-let preExistingObjectsch2 = Map.make(): (Map.Map<Number, (Number, Number)>)
+let preExistingObjectsch2 = Map.make(): Map.Map<Number, (Number, Number)>
 
 let preExistingObjectsWithRefCountMismatch2 = Map.make():
-  (Map.Map<
-    Number,
-    (
+  Map.Map<
       Number,
```
Comment (phated) [inline compiler/test/formatter_outputs/application.gr:65]: Indentation now looks wrong on this.

File: stdlib/runtime/exception.gr [hunk 2]
```diff
@@ -24,10 +24,11 @@ export let dangerouslyRegisterBasePrinter = f => {
...
     // There will be at least one printer registered by the time this is called
-    let (_, next) = WasmI32.toGrain(current): (
-      Exception -> Option<String>,
-      WasmI32
-    )
+    let (_, next) = WasmI32.toGrain(current):
+      (
+          Exception -> Option<String>,
+          WasmI32
+        )
```
Comment (phated) [inline stdlib/runtime/exception.gr:31]: Weird indent

File: stdlib/runtime/exception.gr [hunk 3]
```diff
@@ -58,10 +59,11 @@ let exceptionToString = (e: Exception) => {
...
     }
-    let (printer, next) = WasmI32.toGrain(current): (
-      Exception -> Option<String>,
-      WasmI32
-    )
+    let (printer, next) = WasmI32.toGrain(current):
+      (
+          Exception -> Option<String>,
+          WasmI32
+        )
```
Comment (phated) [inline stdlib/runtime/exception.gr:66]: Weird indent

File: stdlib/runtime/numberUtils.gr [hunk 4]
```diff
@@ -1104,13 +1104,11 @@ let grisuRound = (buffer, len, delta, rest, ten_kappa, wp_w) => {
...
     WasmI64.geU(WasmI64.sub(delta, rest), ten_kappa) &&
-    (
-      WasmI64.ltU(WasmI64.add(rest, ten_kappa), wp_w) ||
+    (WasmI64.ltU(WasmI64.add(rest, ten_kappa), wp_w) ||
       WasmI64.gtU(
         WasmI64.sub(wp_w, rest),
         WasmI64.sub(WasmI64.add(rest, ten_kappa), wp_w)
-      )
-    )
+      ))
```
Comment (phated) [inline stdlib/runtime/numberUtils.gr:1111]: Did this get un-done?

File: stdlib/runtime/numbers.gr [hunk 5]
```diff
@@ -1488,9 +1488,8 @@ let convertInexactToExactHelp = x => {
 }
 
 export let rec convertInexactToExact = (x: Number) => {
-  let ret = WasmI32.toGrain(
-    convertInexactToExactHelp(WasmI32.fromGrain(x))
-  ): Number
+  let ret = WasmI32.toGrain(convertInexactToExactHelp(WasmI32.fromGrain(x))):
+    Number
```
Comment (phated) [inline stdlib/runtime/numbers.gr:1492]: Wonder if we can group this better

**Comment by ospencer** [comment]
@marcusroberts please help 😂

**Comment by marcusroberts** [comment]
> @marcusroberts please help 😂

Marcus has left the building. :-D

**Review by marcusroberts — APPROVED** [review]
Good fix!

File: compiler/test/formatter_outputs/application.gr [hunk 6]
```diff
@@ -53,27 +53,33 @@ let ret = filteri(
...
 
-let preEmatch = Map.make(): (Map.Map<Number, (Number, Number, Number, Number)>)
+let preEmatch = Map.make(): Map.Map<Number, (Number, Number, Number, Number)>
 
-let preExistingObjectsch2 = Map.make(): (Map.Map<Number, (Number, Number)>)
+let preExistingObjectsch2 = Map.make(): Map.Map<Number, (Number, Number)>
 
-let preExistingObjectsWithRefCountMismatch2 = Map.make():
-  (Map.Map<
+let preExistingObjectsWithRefCountMismatch2 = Map.make(): Map.Map<
```
Comment (phated) [inline compiler/test/formatter_outputs/application.gr:70]: Not a huge fan of this, but I hate it **less** than when the type always wraps.

**Review by jozanza — APPROVED** [review]

**Pull request merged**